tests/resources/Group23-VIC-Machine-Service-Util.robot

```diff
@@ -14,21 +14,31 @@
 
 *** Settings ***
 Documentation    This resource contains keywords which are helpful for using curl to test the vic-machine API.
-
+Library  Process
 
 *** Variables ***
-${HTTP_PORT}     1337
-${HTTPS_PORT}    31337
-
 ${RC}            The return code of the last curl invocation
 ${OUTPUT}        The output of the last curl invocation
 ${STATUS}        The HTTP status of the last curl invocation
 
 
 *** Keywords ***
 Start VIC Machine Server
-    Start Process    ./bin/vic-machine-server --port ${HTTP_PORT} --scheme http    shell=True    cwd=/go/src/github.com/vmware/vic
-
+    ${http_port}=    Get HTTP Port
+    Set Suite Variable    ${HTTP_PORT}    ${http_port}
+    ${handle}=    Start Process    ./bin/vic-machine-server --port ${HTTP_PORT} --scheme http    shell=True    cwd=/go/src/github.com/vmware/vic
+    Process Should Be Running    ${handle}
+    [Return]    ${handle}
+
+Stop VIC Machine Server
+    [Arguments]    ${handle}
+    Terminate Process    ${handle}    kill=true
+    Process Should Be Stopped    ${handle}
+
+# Get unused unprivileged TCP port
+Get HTTP Port
+    ${rc}  ${http_port}=    Run And Return Rc And Output    netstat -atn | perl -0777 -ne '@ports = /tcp.*?\:(\d+)\s+/imsg ; for $port (32768..61000) {if(!grep(/^$port$/, @ports)) { print $port; last } }'
```

Comment:
It might be easier to run the server without specifying a port; it will pick an available one at random.

Reply:
@zjs Are you 100% sure?  Because that would be better than this implementation.

Reply:
It definitely picks a random one.

That said, I'm not sure of the best way to see what port was picked.

Currently, I think it's just logged. We're not currently specifying a unique logfile for each invocation, but it seems like that would be easy enough to do (e.g., by writing to a file in `${OUTPUT DIR}`).

If we do that, we could grep for a line like `Serving vic machine at http://127.0.0.1:40081` and get the port number off of the end. (Or, perhaps even better, just stick the entire URL in a variable and use that in the test code instead of using the port number with a hard-coded protocol and host, making it easier to add tests for HTTPS in the future.)

Reply:
I will get this going.  Thanks @zjs!